seispy/image.py

```diff
@@ -0,0 +1,402 @@
+        else:
+            for i in patches:
+                self.patches[i].plot(*args, **kwargs)
+
+    @property
+    def is_cross_section(self):
+        if self._plane in (0, 1):
+            return True
+        elif self._plane == 2:
+            return False
+
+    @property
+    def number_of_patches(self):
+        return len(self.patches)
+
+    @property
+    def precision(self):
+        return SW4_IMAGE_PRECISION[self._precision]
+
+    @property
+    def plane(self):
+        return SW4_IMAGE_PLANE[self._plane]
+
+    @property
+    def type(self):
+        if self.is_cross_section:
+            return 'cross-section'
+        else:
+            return 'map'
+
+    @property
+    def quantity_name(self):
+        return self._mode_dict[self._mode]['name']
+
+    @property
+    def quantity_symbol(self):
+        return self._mode_dict[self._mode]['symbol']
+
+    @property
+    def quantity_unit(self):
+        return self._mode_dict[self._mode]['unit']
+
+
+class Patch(object):
+    """
+    A class to hold WPP or SW4 patch data
+    """
+
+    def __init__(self, number=None, image=None):
+        self.number = number
+        self._image = image  # link back to the image this patch belongs to
+        self.h = None
+        self.zmin = None
+        self.ib = None
+        self.ni = None
+        self.jb = None
+        self.nj = None
+        self.data = None
+        self.extent = None
+
+    def _set_header(self, header):
+        """
+        Set SW4 patch header information
+        """
+        (self.h,
+         self.zmin,
+         self.ib,
+         self.ni,
+         self.jb,
+         self.nj) = header
+
+    def _set_data(self, data):
+        self.data = data
+        if self._image.is_cross_section:
+            self.extent = (
+                0 - (self.h / 2.0),
+                (self.ni - 1) * self.h + (self.h / 2.0),
+                self.zmin - (self.h / 2.0),
+                self.zmin + (self.nj - 1) * self.h + (self.h / 2.0))
+        else:
+            self.data = self.data.T
+            self.extent = (
+                0 - (self.h / 2.0),
+                (self.nj - 1) * self.h + (self.h / 2.0),
+                0 - (self.h / 2.0),
+                (self.ni - 1) * self.h + (self.h / 2.0))
+        self.min = data.min()
+        self.max = data.max()
+        self.std = data.std()
+        self.rms = np.sqrt(np.mean(data**2))
+
+    def plot(self, ax=None, vmin='min', vmax='max', colorbar=True,
+             colorbar_label=None, cmap=None, **kwargs):
+
+        if ax is None:
+            fig, ax = plt.subplots()
+
+        ax.set_aspect(1)
+
+        if vmax is 'max':
+            vmax = self.max
+        elif type(vmax) is str:
+            try:
+                factor = float(vmax)
+                vmax = factor*self.rms
+                if self.min < 0:
+                    vmin = -vmax
+
+            except ValueError:
+                print ('Warning! keyword vmax=$s in not understood...\n' %vmax,
```

Comment:
Hmm, actually I removed that part, since it seemed a bit too cryptic too me.. if this is really used functionality, maybe we can add it more programmatically again later?